### DISCUSSION TITLE
Creating a page

### DISCUSSION THREAD

**Discussion by usmcgator** [filamentphp/filament#5621] [Help]
I'm pretty new to TALL stack and have been making good progress using the admin panel. I needed to create a basic page that was not tied to a model as it only needs to make an external API call to a 3rd party and display data on the page. The Filament documentation for creating a basic page only provides enough info on how to create a blank page with a nav item and a title. I tried using the render method to pass data to a view, but it said the render method is not compatible with Filament pages. Is there an example of how to pass data to a page view? Thanks.

**Comment by parijke** [comment]
I think Ryan is doing something like that here ?
https://youtu.be/Sr15MWxNfrA?t=4878

**Comment by dave-mills** [comment]
I've been trying to work this out too. This is where I've got to - maybe it'll help!

The `Filament\Pages\Page` class has a render function that references `$this->getViewData()`. So, if you're using a custom page that extends `Filament\Pages\Page`, you can override that method and return an array of any data you like. E.g: 


```php

namespace App\Filament\Pages;

use Filament\Pages\Page;

class Custom extends Page
{
    protected static ?string $navigationIcon = 'heroicon-o-document-text';
    protected static string $view = 'filament.pages.custom';

    public function getViewData(): array
    {
        return [
            'test' => 'your API data is ready',
            'foo'  => 'bar',
        ];
    }
}
```

Then the blade template: 

```
<x-filament::page>

    Test data = {{ $test }}
    <br/>
    Foo = {{ $foo }}
    
</x-filament::page>
```

... No idea if this is the best way to do it, but it works for me!

**Reply by usmcgator** [reply]
That's exactly what I needed - worked perfectly.  Thanks!!!
